common/src/service/HttpClientHelper.cs

```diff
@@ -66,7 +66,17 @@ sealed class HttpClientHelper : IHttpClientHelper
             this.httpClientObj.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue(CommonConstants.MediaTypeForDeviceManagementApis));
             this.httpClientObj.DefaultRequestHeaders.ExpectContinue = false;
 
-            this.httpClientObjWithPerRequestTimeout = new HttpClient();
+            if (customHttpProxy != DefaultWebProxySettings.Instance)
```

Comment:
@abhipsaMisra @jenoola please work together to identify what is missing from the existing serviceclient code: we already have proxy support for ServiceClient C2D (test should exist). as well as RegistryManager, JobClient, etc which work on HTTP. 

Needing to add this code here means we either missed something or the previous code was not added at the right depth in our stack.